chromadb/test/auth/test_simple_rbac_authz.py

```diff
@@ -0,0 +1,256 @@
+        col := Collection(api, f"test-delete-doc-{uuid.uuid4()}", uuid.uuid4()),
+        col.delete(ids=["1"]),  # type: ignore
+    ),
+    "collection:get": lambda api: (
+        col := Collection(api, f"test-get-docs-{uuid.uuid4()}", uuid.uuid4()),
+        col.get(ids=["1"]),
+    ),
+    "collection:query": lambda api: (
+        col := Collection(api, f"test-query-{uuid.uuid4()}", uuid.uuid4()),
+        col.query(query_texts=["test"]),
+    ),
+    "collection:peek": lambda api: (
+        col := Collection(api, f"test-peek-{uuid.uuid4()}", uuid.uuid4()),
+        col.peek(),
+    ),
+    "collection:update": lambda api: (
+        col := Collection(api, f"test-update-docs-{uuid.uuid4()}", uuid.uuid4()),
+        col.update(ids=["1"], documents=["test1"]),  # type: ignore
+    ),
+    "collection:upsert": lambda api: (
+        col := Collection(api, f"test-upsert-{uuid.uuid4()}", uuid.uuid4()),
+        col.upsert(ids=["1"], documents=["test1"]),  # type: ignore
+    ),
+    "collection:count": lambda api: (
+        col := Collection(api, f"test-count-{uuid.uuid4()}", uuid.uuid4()),
+        col.count(),
+    ),
+}
+
+
+@settings(max_examples=10)
+@given(token_config=token_config(), rbac_config=rbac_config())
+def test_authz(token_config: Dict[str, Any], rbac_config: Dict[str, Any]) -> None:
+    authz_config = rbac_config
+    token_config["chroma_server_authz_config"] = rbac_config
+    token_config["chroma_server_auth_credentials"] = json.dumps(rbac_config["users"])
+    random_user = random.choice(authz_config["users"])
+    random_token = random.choice(random_user["tokens"])["token"]
+    api = _fastapi_fixture(
+        is_persistent=token_config["is_persistent"],
+        chroma_server_auth_provider=token_config["chroma_server_auth_provider"],
+        chroma_server_auth_credentials_provider=token_config[
+            "chroma_server_auth_credentials_provider"
+        ],
+        chroma_server_auth_credentials=token_config["chroma_server_auth_credentials"],
+        chroma_client_auth_provider=token_config["chroma_client_auth_provider"],
+        chroma_client_auth_token_transport_header=token_config[
+            "token_transport_header"
+        ],
+        chroma_server_auth_token_transport_header=token_config[
+            "token_transport_header"
+        ],
+        chroma_server_authz_provider=token_config["chroma_server_authz_provider"],
+        chroma_server_authz_config=token_config["chroma_server_authz_config"],
+        chroma_client_auth_credentials=random_token,
+    )
+    _sys: System = next(api)
+    _sys.reset_state()
+    _api = _sys.instance(ServerAPI)
+    _api.heartbeat()
+    for actions in authz_config["roles_mapping"][random_user["role"]]["actions"]:
+        try:
+            print(actions)
+            api_executors[actions](_api)  # type: ignore
+        except Exception as e:
+            # we want to ignore errors such as collection not found or 400 client error
```

Comment:
Lets make a master role that can write data instead

Reply:
Done with that, testing now